core/imt/test/testEnableImt.cxx

```diff
@@ -0,0 +1,26 @@
+#include "TROOT.h"
+#include "ROOT/RTaskArena.hxx"
+
+#include "tbb/task_arena.h"
+
+#include "ROOT/TestSupport.hxx"
+#include "gtest/gtest.h"
+
+#ifdef R__USE_IMT
+
+static const unsigned gMaxConcurrency = ROOT::Internal::LogicalCPUBandwidthControl();
+
+TEST(EnableImt, TBBAttach)
+{
+   tbb::task_arena main_arena;
+   main_arena.initialize(2);
+
+   ROOT::EnableImplicitMT(ROOT::EIMTConfig::kExistingTBBArena);
```

Comment:
Just creating the arena is insufficient, you must also use it
```C++
main_arena.execute([&]() {
 ROOT::EnableImplicitMT(ROOT::EIMTConfig::kExistingTBBArena);
   auto psize = ROOT::GetThreadPoolSize();

   EXPECT_TRUE( psize > 1);
   EXPECT_EQ( psize, 2);
});
```

Reply:
> you must also use it

To be more specific, I need to have the ImplicitMT initialization done within a call to the area.
```
main_arena.execute([&]() {
    ROOT::EnableImplicitMT(ROOT::EIMTConfig::kExistingTBBArena);
});

 auto psize = ROOT::GetThreadPoolSize();

 EXPECT_TRUE( psize > 1);
 EXPECT_EQ( psize, 2);
```
succeeds.  'Just' using the main_arena and then calling `EnableImplicitMT` on the main thread does not 'work.

Reply:
Just as a quick note, I do not believe the code in `ROOT::GetThreadPoolSize();` is ultimately thread safe. It appears to just be a member `static` of `RTaskArenaWrapper` without any (obvious) thread synchronization.